File: starters/dotnet/FProto.Core/TransportType.cs
namespace FProto.Core;

public enum TransportType
{
    WebSocket,
    Quic,
    Grpc
}

public interface IConn
{
    Task<byte[]> ReadMessageAsync(CancellationToken ct = default);
    Task WriteMessageAsync(byte[] data, CancellationToken ct = default);
    Task CloseAsync(CancellationToken ct = default);
    TransportType TransportType { get; }
    string RemoteAddr { get; }
}

File: starters/java/src/main/java/io/fproto/core/TransportType.java
package io.fproto.core;

public enum TransportType {
    WEBSOCKET("websocket"),
    QUIC("quic"),
    GRPC("grpc");

    private final String value;

    TransportType(String value) {
        this.value = value;
    }

    public String getValue() {
        return value;
    }

    @Override
    public String toString() {
        return value;
    }
}
